tiledb/ml/models/tensorflow_keras.py

```diff
@@ -200,102 +242,128 @@ def load(
-        else:
-            fields = [
-                "weight_names",
-                "weight_values",
-                "layer_name",
-                "optimizer_weights",
-            ]
-        super()._create_array(domain_info, fields)
-
-    def _write_array(
-        self,
-        include_optimizer: bool,
-        serialized_weights: bytes,
-        serialized_optimizer_weights: bytes,
-        meta: Optional[Meta],
-    ) -> None:
-        """Write Tensorflow model to a TileDB array."""
-        assert self.artifact
-        # TODO: Change timestamp when issue in core is resolved
-        with tiledb.open(
-            self.uri, "w", timestamp=current_milli_time(), ctx=self.ctx
-        ) as tf_model_tiledb:
-            if isinstance(self.artifact, FunctionalOrSequential):
-                tf_model_tiledb[:] = {
-                    "model_weights": np.array([serialized_weights]),
-                    "optimizer_weights": np.array([serialized_optimizer_weights]),
-                }
-            else:
-                # Insert weights and optimizer
-                layer_names = []
-                weight_names = []
-                weight_values = []
-                for layer in sorted(self.artifact.layers, key=attrgetter("name")):
-                    weights = layer.trainable_weights + layer.non_trainable_weights
-                    weight_values.append(
-                        pickle.dumps(tf.keras.backend.batch_get_value(weights))
-                    )
-                    weight_names.append(
-                        pickle.dumps([w.name.encode("utf8") for w in weights])
-                    )
-                    layer_names.append(layer.name)
-
-                tf_model_tiledb[:] = {
-                    "layer_name": np.array(layer_names),
-                    "weight_values": np.array(weight_values),
-                    "weight_names": np.array(weight_names),
-                    # TODO (TeamML) Fix array scheme to avoid optimizer_weight repetition. Nullable
-                    "optimizer_weights": np.array(
-                        [
-                            serialized_optimizer_weights
-                            for _ in range(len(self.artifact.layers))
-                        ]
-                    ),
-                }
-
-            # Insert all model metadata
-            model_metadata = saving_utils.model_metadata(
-                self.artifact, include_optimizer
-            )
-            for key, value in model_metadata.items():
-                tf_model_tiledb.meta[key] = json.dumps(
-                    value, default=get_json_type
-                ).encode("utf8")
+    def get_weights(self, timestamp: Optional[Timestamp] = None) -> List[np.ndarray]:
+        """
+        Returns the weights of a Tensorflow Keras model.
+        """
+        with tiledb.open(self.uri, ctx=self.ctx, timestamp=timestamp) as model_array:
+            model_meta = dict(model_array.meta.items())
```

Comment:
No need to copy model_array.meta to a new dict, it supports the necessary dict methods